nodebuilder/tests/sync_test.go

```diff
@@ -17,61 +15,184 @@ import (
 
 // Common consts for tests producing filled blocks
 const (
-	blocks = 20
-	bsize  = 16
-	btime  = time.Millisecond * 300
+	numBlocks = 20
+	bsize     = 16
+	btime     = time.Millisecond * 300
 )
 
 /*
-Test-Case: Sync a Light Node with a Bridge Node(includes DASing of non-empty blocks)
+Test-Case: Header and block/sample sync against a Bridge Node of non-empty blocks.
+
 Steps:
 1. Create a Bridge Node(BN)
 2. Start a BN
 3. Check BN is synced to height 20
-4. Create a Light Node(LN) with a trusted peer
+
+Light node:
+4. Create a Light Node (LN) with bridge as a trusted peer
 5. Start a LN with a defined connection to the BN
-6. Check LN is synced to height 30
+6. Check LN is header-synced to height 20
+7. Wait until LN has sampled height 20
+8. Wait for LN DASer to catch up to network head
+
+Full node:
+4. Create a Full Node (FN) with bridge as a trusted peer
+5. Start a FN with a defined connection to the BN
+6. Check FN is header-synced to height 20
+7. Wait until FN has synced block at height 20
+8. Wait for FN DASer to catch up to network head
 */
-func TestSyncLightWithBridge(t *testing.T) {
+func TestSyncAgainstBridge_NonEmptyChain(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), swamp.DefaultTestTimeout)
 	t.Cleanup(cancel)
 
 	sw := swamp.NewSwamp(t, swamp.WithBlockTime(btime))
-	fillDn := swamp.FillBlocks(ctx, sw.ClientContext, sw.Accounts, bsize, blocks)
+	// wait for core network to fill 20 blocks
+	fillDn := swamp.FillBlocks(ctx, sw.ClientContext, sw.Accounts, bsize, numBlocks)
+	sw.WaitTillHeight(ctx, numBlocks)
 
+	// start a bridge and wait for it to sync to 20
 	bridge := sw.NewBridgeNode()
-
-	sw.WaitTillHeight(ctx, 20)
-
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
 
-	h, err := bridge.HeaderServ.WaitForHeight(ctx, 20)
+	h, err := bridge.HeaderServ.WaitForHeight(ctx, numBlocks)
 	require.NoError(t, err)
+	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, numBlocks))
+
+	t.Run("light sync against bridge", func(t *testing.T) {
```

Comment:
Suggestion: Looks like at this stage, blocks and headers are already produced by BN and there the Light and Full node only need to sync. Maybe we can use lower context, that is local for each `t.Run` to allow faster end if tests fails.

Reply:
Hmm why lower the context? We still need to wait for DASing too.